### DISCUSSION TITLE
Can we have parmeterized action methods?

### DISCUSSION THREAD

**Discussion by bjmc** [pmndrs/zustand#2390] [Q&A]
Hello, we've got an application that's going to need a great number of actions to update small pieces of a large complicated state object.

I was wondering about ways to reduce boilerplate and avoid writing a zillion actions like `updateLeftSock()`, `updateRightSock()`, `updateLeftShoelace()`... 

It occurred to me that maybe it would be possible to create a parameterized update action that takes a [JSONPath](https://www.toolsqa.com/rest-assured/jsonpath-and-query-json-using-jsonpath/) expression, and a new value and applies the appropriate changes.

In pseudocode, I'm imagining something like following:

```typescript
immer(...
  updateFromPath(path: JSONPath, newValue) => {
    set(state) => {
      magicJsonPathSetter(state, path, newValue);
    }
})
```

I don't have a working example (clearly) but I thought I'd ask here before I went off down a blind alley. Are there reasons it's difficult or impossible to do what I'm imagining? Does this contradict how Immer or Zustand internals work?

Has anyone done this already, and have a working example they'd be able to share? Thanks!

**Comment by dbritto-dev** [comment]
@bjmc there's no issue about using immer and zustand, you can even don't have actions and just call `store.setState()` and that would be fine
